src/main/java/lol/ui/Sprites.java

```diff
@@ -2,24 +2,50 @@
 
 import javafx.scene.image.Image;
 import javafx.scene.image.ImageView;
+import lol.game.Nexus;
 
 public class Sprites {
   private Image grassImage;
-  private Image archerImage;
-  private Image warriorImage;
+  private Image redArcherImage;
+  private Image blueArcherImage;
+  private Image redWarriorImage;
+  private Image blueWarriorImage;
   private Image blueNexusImage;
   private Image redNexusImage;
   private Image blueTowerImage;
   private Image redTowerImage;
+  private Image treeImage;
+  private Image rockImage;
+  private Image[] archerImages = new Image[2];
+  private Image[] warriorImages = new Image[2];
+  private Image[] towerImages = new Image[2];
+  private Image[] nexusImages = new Image[2];
 
   public Sprites() {
     grassImage = new Image("sprites/grass-tile.png");
-    archerImage = new Image("sprites/archer.png");
-    warriorImage = new Image("sprites/warrior.png");
+    rockImage = new Image("sprites/rock-tile.png");
+    treeImage = new Image("sprites/tree-tile.png");
+
+    blueArcherImage = new Image("sprites/blue-archer.png");
+    redArcherImage = new Image("sprites/red-archer.png");
+    archerImages[Nexus.BLUE] = blueArcherImage;
```

Comment:
you should remove all the images `blueArcher...`, `redArcher`, ...

Instead:

```java
archerImages[Nexus.BLUE] = new Image("sprites/blue-archer.png");
...
```

and similarly for all the others.